src/Symfony/Component/Validator/Constraints/Email.php

```diff
@@ -31,8 +35,37 @@ class Email extends Constraint
         self::HOST_CHECK_FAILED_ERROR => 'HOST_CHECK_FAILED_ERROR',
     );
 
+    /**
+     * @var string[]
+     *
+     * @internal
+     */
+    public static $validationModes = array(
+        self::VALIDATION_MODE_HTML5,
+        self::VALIDATION_MODE_STRICT,
+        self::VALIDATION_MODE_LOOSE,
+    );
+
     public $message = 'This value is not a valid email address.';
     public $checkMX = false;
     public $checkHost = false;
+
+    /**
+     * @deprecated since version 3.4, to be removed in 4.0. Set mode to "strict" instead.
```

Comment:
`@deprecated since version 4.1, to be removed in 5.0. Set mode to "strict" instead.`

Reply:
The other messages need to be updated accordingly.